src/Apps/ArtAdvisor/07-Curated-Discovery/weaviate-db.ts

```diff
@@ -60,6 +66,51 @@ export class WeaviateDB {
 
     return result as DiscoveryMarketingCollections[]
   }
+
+  /**
+   * Fetch articles by semantic search on an array of concepts.
+   */
+  async getNearArticles({
+    concepts,
+    limit = 10,
+  }: {
+    /** List of concepts for semantic search */
+    concepts: string[]
+    /** Max number of articles to return */
+    limit?: number
+  }) {
+    console.log("[WeaviateDB] getNearArticles", {
+      concepts,
+      limit,
+    })
+
+    const conceptArray = ensureValidConcepts(concepts)
+
+    const response = await this.client.graphql
+      .get()
+      .withClassName(this.articlesClass)
+      .withNearText({
+        concepts: conceptArray,
+      })
+      .withLimit(limit)
+      .withFields(
+        "internalID title articleDescription href imageUrl _additional { id distance }"
+      )
+      .do()
+
+    const result: ArticleType[] = response.data.Get[DEFAULT_ARTICLES_CLASS]
+
+    /**
+     * Note: Because we return article sections, a single article can appear
+     * multiple times in the results. This removes duplicates based on the
+     * internalID. Order is preserved and the first occurrence is kept.
+     *
+     * TODO: handle better?
+     */
+    const dedupedResults = _.uniqBy(result, "internalID")
```

Comment:
Good call — I wonder if this means we might want to overfetch (i.e. more than 10 sections), in order to reduce the chance of getting puny results because of the deduping.